### PR TITLE
Remove unnecessary machine specific client

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR removes unnecessary machine specific client, it can be replaced with controller runtime client.

### CODE REVIEW

**Comment by enxebre** [comment]
can we please add a PR description with the motivation of this PR?

`ci/prow/git-history` requires commit format to be `UPSTREAM: <carry>: openshift: commit message`

All worker machines here are stuck in provisioned phases https://gcsweb-ci.svc.ci.openshift.org/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-api-provider-azure/98/pull-ci-openshift-cluster-api-provider-azure-master-e2e-azure-operator/203/artifacts/e2e-azure-operator/must-gather/registry-svc-ci-openshift-org-ci-op-bhz0741x-stable-sha256-dd03a9e0ac7b6e710037d8f0d0a5001b47c27bf859e49b0256e8ac54f5bd8198/namespaces/openshift-machine-api/machine.openshift.io/machines/

have you had the chance to validate the change locally?

**Comment by enxebre** [comment]
/approve

@bison PTAL
@alexander-demichev can you please put a follow up to consolidate with this approach https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/machine_scope.go#L88 and use patch requests instead?

File: pkg/cloud/azure/actuators/machine_scope.go [hunk 1]
```diff
@@ -109,8 +109,9 @@ func NewMachineScope(params MachineScopeParams) (*MachineScope, error) {
 		MachineStatus: machineStatus,
 		// Once set, they can not be changed. Otherwise, status change computation
 		// might be invalid and result in skipping the status update.
-		origMachine:       params.Machine.DeepCopy(),
-		origMachineStatus: machineStatus.DeepCopy(),
+		origMachine:        params.Machine.DeepCopy(),
```
Comment (enxebre) [inline pkg/cloud/azure/actuators/machine_scope.go]: can we stop deepcopying here

File: pkg/cloud/azure/actuators/machine_scope_test.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ func init() {
 }
 
 func providerSpecFromMachine(in *machineproviderv1.AzureMachineProviderSpec) (*machinev1.ProviderSpec, error) {
-	bytes, err := yaml.Marshal(in)
+	bytes, err := json.Marshal(in)
```
Comment (enxebre) [inline pkg/cloud/azure/actuators/machine_scope_test.go]: why?

**Comment by enxebre** [comment]
In a follow up can we drop origMachine and origProviderStatus and keep consolidating impl from aws/gcp/azure/vsphere?

File: pkg/cloud/azure/actuators/machine_scope_test.go (hunk 2 above)
Comment (alexander-demichev) [reply pkg/cloud/azure/actuators/machine_scope_test.go]: Because `.Patch()` expects it to be JSON and fails if it's yaml

**Pull request merged**